tinyxml2.h

```diff
@@ -190,7 +191,7 @@ class StrPair
     char*   _end;
 
     StrPair( const StrPair& other );	// not supported
-    void operator=( StrPair& other );	// not supported, use TransferTo()
+    StrPair& operator=( const StrPair& other );	// not supported, use TransferTo()
 };

```

Comment:
This makes things worse. Before this change you would write `a=b=c` and the compiler would tell you that this wouldn't work because of wrong signature. After this change you would be able to write `a=b=c` and it would complie and only linkage would fail.

Reply:
The only problem with the `a=b=c` is this can _only_ happen within `StrPair` methods and friends of `StrPair`. All external users of the class are already "protected" by it being private.
Another issus is the lack of `const`ness of `other`. An additional method with the interface `... operator=( const StrPair& other );` can be added and coexist with `void operator=( StrPair& other );`.
Keeping interfaces recognizable by code reviews and analysis tools also helps and keeps it consistent with the other interfaces within this particular file.

Reply:
I completely missed the fact that the parameter had the wrong type - should be "reference to const object" but it's actually "reference to non-const object". Could you please submit the parameter type change alone as a separate PR?

Reply:
No problem, Checkout issue #718